### PR TITLE
Migrate image-pruner CronJob to a Periodic Prowjob

### PR DESCRIPTION
**Pull request by petr-muller**
...and move its supporting config from Makefile to `core-services`.

/hold

I am not entirely sure if I adapted the former `oc adm policy add-cluster-role-to-user system:image-pruner -z image-pruner` command from the Makefile correctly in in `core-services/image-pruner/admin_rbac.yaml`. The command seems to modify an existing `system:image-pruner` CRB, but I did not want to hold config for a system CRB in a random directory so I created a new CRB referring to the same role.

The README.md could also use some context about why we are running this job.

/cc @stevekuznetsov @smarterclayton @bbguimaraes @droslean @hongkailiu

### CODE REVIEW

File: core-services/image-pruner/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Image Pruner
+
+Configuration supporting the [image pruning job](https://prow.svc.ci.openshift.org/?job=periodic-ci-image-pruner).
+
+TODO: Could use some background. Why are we pruning images?
```
Comment (stevekuznetsov) [inline core-services/image-pruner/README.md]: We create some ludicrous amount of images in every image build for the build farm, if we don't prune them we're just using up terabytes of space holding layers nobody will ever reference again.

File: core-services/image-pruner/admin_rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+kind: List
+apiVersion: v1
+items:
+- apiVersion: v1
+  kind: ServiceAccount
+  metadata:
+    name: image-pruner
+    namespace: ci
+- apiVersion: authorization.openshift.io/v1
+  kind: ClusterRoleBinding
```
Comment (droslean) [inline core-services/image-pruner/admin_rbac.yaml]: I the current clusterrolebinding there is also:
```
userNames:
- system:serviceaccount:ci:image-pruner
```
I am not sure if that will be added automatically, but it worth taking a look.

Comment (petr-muller) [reply core-services/image-pruner/admin_rbac.yaml]: Yeah, I wondered about that too. I have found nothing about that on https://kubernetes.io/docs/reference/access-authn-authz/rbac/ or https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.15/#clusterrolebinding-v1-rbac-authorization-k8s-io, so it really looks like something that will be put there automatically
/shrug

**Comment by petr-muller** [comment]
Added some context to README

File: core-services/image-pruner/admin_rbac.yaml (hunk 2 above)
Comment (stevekuznetsov) [reply core-services/image-pruner/admin_rbac.yaml]: Yeah it gets generated

**Pull request merged**